vortex-scalar/src/struct_.rs

```diff
@@ -80,25 +80,19 @@ impl<'a> StructScalar<'a> {
     }
 
     pub fn field_by_idx(&self, idx: usize) -> VortexResult<Scalar> {
-        let DType::Struct(st, nullability) = self.dtype() else {
-            unreachable!()
-        };
-
-        let mut field_dtype = st.field_by_index(idx)?;
-        if matches!(nullability, Nullability::Nullable) {
-            field_dtype = field_dtype.with_nullability(Nullability::Nullable);
-        }
-
-        Ok(match self.fields {
-            None => Scalar::null(field_dtype),
+        match self.fields {
+            None => vortex_panic!("Can't take field {idx} out of null struct scalar"),
```

Comment:
You could `.vortex_expect` right? Nit-picking!

Reply:
yeah, less wrapping